src/mongoc/mongoc-write-command.c

```diff
@@ -109,6 +107,71 @@ _mongoc_write_command_insert_append (mongoc_write_command_t *command,
    EXIT;
 }
 
+void
+_mongoc_write_command_update_append (mongoc_write_command_t *command,
+                                     const bson_t* selector,
+                                     const bson_t* update,
+                                     bool          upsert,
+                                     bool          multi)
+{
+   char id[20];
+   bson_t doc;
+   uint32_t pos;
+
+   ENTRY;
+
+   BSON_ASSERT (command);
+   BSON_ASSERT (command->type == MONGOC_WRITE_COMMAND_UPDATE);
+   BSON_ASSERT (selector && update);
+
+   bson_init(&doc);
+   BSON_APPEND_DOCUMENT (&doc, "q", selector);
+   BSON_APPEND_DOCUMENT (&doc, "u", update);
+   BSON_APPEND_BOOL (&doc, "upsert", upsert);
+   BSON_APPEND_BOOL (&doc, "multi", multi);
+
+   pos = command->u.update.n_updates;
+   sprintf(id, "%i", pos);
```

Comment:
There's actually a bson_uint32_to_string() function which we've been preferring over sprintf for these kind of conversions